### PR TITLE
sui: testnet support

### PR DESCRIPTION
**Pull request by evan-gray**

### CODE REVIEW

File: src/hooks/useGetIsTransferCompleted.ts [hunk 1]
```diff
@@ -240,6 +243,24 @@ export default function useGetIsTransferCompleted(
             setIsLoading(false);
           }
         })();
+      } else if (targetChain === CHAIN_ID_SUI) {
+        setIsLoading(true);
```
Comment (M-Picco) [inline src/hooks/useGetIsTransferCompleted.ts:247]: Is it possible there's a `setIsLoading(false)` missing in case of an error? I noticed it's missing for other chains too, so it'd be better to leave it for another PR/issue discussion.

File: src/hooks/useHandleCreateWrapped.tsx [hunk 2]
```diff
@@ -82,6 +86,76 @@ import { useTerraWallet } from "../contexts/TerraWalletContext";
...
+  getWrappedCoinType,
+} from "@certusone/wormhole-sdk/lib/cjs/sui";
+import { getSuiProvider } from "../utils/sui";
+import { sleep } from "../utils/sleep";
+import { useSuiWallet } from "../contexts/SuiWalletContext";
+import { SuiWallet } from "@xlabs-libs/wallet-aggregator-sui";
+import { createWrappedOnSuiPrepare } from "../utils/suiPublishHotfix";
+
+// TODO: replace with SDK method
+export async function updateWrappedOnSui(
```
Comment (M-Picco) [inline src/hooks/useHandleCreateWrapped.tsx:106]: Is this method already in the sdk (see [here](https://github.com/wormhole-foundation/wormhole/blob/sui/integration_v2/sdk/js/src/token_bridge/updateWrapped.ts#L45))? I don't know which version the bet package was built from though, so this might not be the case

File: src/hooks/useHandleCreateWrapped.tsx [hunk 3]
```diff
@@ -421,7 +495,136 @@ async function injective(
...
+          transactionBlock: suiPrepareRegistrationTxPayload,
+          options: {
+            showObjectChanges: true,
+          },
+        })
+      ).data;
+      if (!suiPrepareRegistrationTxRes) {
+        throw new Error("Error parsing transaction results");
+      }
+      console.log(suiPrepareRegistrationTxRes);
```
Comment (M-Picco) [inline src/hooks/useHandleCreateWrapped.tsx:549]: Remove this `console.log`

File: src/utils/suiPublishHotfix.ts [hunk 4]
```diff
@@ -0,0 +1,103 @@
...
+} from "@certusone/wormhole-sdk/lib/cjs/sui";
+import { SuiBuildOutput } from "@certusone/wormhole-sdk/lib/cjs/sui/types";
+import {
+  fromB64,
+  JsonRpcProvider,
+  normalizeSuiObjectId,
+  TransactionBlock,
+} from "@mysten/sui.js";
+
+export async function createWrappedOnSuiPrepare(
```
Comment (M-Picco) [inline src/utils/suiPublishHotfix.ts:13]: Just for the record: what would the hotfix be for?

**Review by M-Picco — APPROVED** [review]
LGTM.

Comments are nitpicks or informational and can be ignored.

File: src/hooks/useGetIsTransferCompleted.ts (hunk 1 above)
Comment (evan-gray) [reply src/hooks/useGetIsTransferCompleted.ts:247]: This is covered by line 261 (and matches the other functions). The error is caught and buried, then if the effect isn't cancelled loading will be set to false.

File: src/hooks/useHandleCreateWrapped.tsx (hunk 2 above)
Comment (evan-gray) [reply src/hooks/useHandleCreateWrapped.tsx:106]: Unfortunately not released yet 😢

File: src/utils/suiPublishHotfix.ts (hunk 4 above)
Comment (evan-gray) [reply src/utils/suiPublishHotfix.ts:13]: added a comment
